File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let ans = 0; 
    let mp = new Set(nums);

    for(let val of mp){
        if(!mp.has(val-1)){
            let newv = val;
            let count = 0;
            while(mp.has(newv)){
                count++;
                newv++;
            }
            ans = Math.max(ans, count);
        }
    }
    return ans;
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        set<int> hashSet(nums.begin(), nums.end());
        int maximum= 0;

        for(auto x : hashSet)
        {
            if(!hashSet.count(x-1))
            {
                int curr_num = x;
                int curr_len = 1;

                while(hashSet.count(curr_num+1))
                {
                    curr_num += 1;
                    curr_len += 1;
                }
                maximum = max(maximum, curr_len);
            }
        }
        return maximum;
        
    }
};